### PR TITLE
feat17: Added functionality to the Modal component

### PR DESCRIPTION
**Pull request by SinchanaMS**
Added functionality to the Modal.

Preview: 

https://user-images.githubusercontent.com/94738532/161756697-84e27636-0568-42ff-8082-70dbda44cdeb.mp4

### CODE REVIEW

File: js/modals.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+var modalCTA1 = document.querySelector(".modal-cta-one")
+var modalCTA2 = document.querySelector(".modal-cta-two")
+
+
+modalHandlerBtn.addEventListener("click", ()=>modalBg.style.display="block")
+modalCTA.addEventListener("click", ()=>modalBg.style.display="none")
+
+modalCentreHandlerBtn.addEventListener("click", ()=>modalCentreBg.style.display="block")
+modalCTA1.addEventListener("click", ()=>modalCentreBg.style.display="none")
+modalCTA2.addEventListener("click", ()=>modalCentreBg.style.display="none")
```
Comment (manojsarna) [inline js/modals.js:15]: Suggestion : just "display:none" the modal container 

` <div class="modal-bg-content-centre">` you need to toggle on this only

File: docs/modals.html [hunk 2]
```diff
@@ -121,17 +124,19 @@ <h1 class="wrapper-heading">Modals - CTA at the centre</h1>
...
-                        <h1>Order Placed!</h1>
-                        <hr />
-                        <p class="p-lg">Congratulations! Your order has been confirmed.</p>
-                        <p class="p-lg">Please check your email for details.</p>
-                        <div>
-                            <button class="modal-cta icon-light">OK</button>
-                            <button class="modal-cta icon-dark"> Order Summary</button>
+                    <div class="modal-bg-content-centre">
+                        <div class="modal">
+                            <h1>Order Placed!</h1>
```
Comment (manojsarna) [inline docs/modals.html]: Suggestion : dont use class names like `"modal"` you will get into trouble with CSS clashes when you use this component in react as common class names will override the CSS, use custom names like `"clay-modal"`

File: stylesheets/components/modals/modal.css [hunk 3]
```diff
@@ -5,30 +5,45 @@
...
+    top: 0;
+    width: 100%;
+    height: 100%;
+    overflow: auto;
+    background-color: var(--MODAL-BG);
+}
+
 .modal {
     display: flex;
     flex-direction: column;
```
Comment (manojsarna) [inline stylesheets/components/modals/modal.css]: Suggestion : Dont use common class names in CSS

**Review by manojsarna — APPROVED** [review]
Overall looks good

DO check the comments

**Review by m26unkwn — APPROVED** [review]
Everything seems good to me and you can checkout the comments given by @manojsarna

File: docs/modals.html [hunk 4]
```diff
@@ -121,17 +124,19 @@ <h1 class="wrapper-heading">Modals - CTA at the centre</h1>
...
+                            <hr />
+                            <p class="p-lg">Congratulations! Your order has been confirmed.</p>
+                            <p class="p-lg">Please check your email for details.</p>
```
Comment (saurabhchirde) [inline docs/modals.html:132]: Good to see that you are using Utility classes.

File: stylesheets/components/modals/modal.css [hunk 5]
```diff
@@ -5,30 +5,45 @@
...
-    width: 60%;
+    width: 50%;
     position: absolute;
     top: 50%;
-    left: 50%;
+    left: 57%;
     transform: translate(-50%, -50%);
     border-radius: 4px;
```
Comment (saurabhchirde) [inline stylesheets/components/modals/modal.css:33]: Good practice of using px unit for border-radius.

**Review by saurabhchirde — APPROVED** [review]
Clean and consistent code practice. Good job.
Looks good to merge

File: docs/modals.html [hunk 6]
```diff
@@ -76,17 +76,20 @@ <h1 class="wrapper-heading">Modals - CTA on the right end</h1>
...
-                        <p class="p-lg">Please check your email for details.</p>
-                        <div class="modal-cta-right">
-                            <button class="modal-cta icon-light">OK</button>
-                            <button class="modal-cta icon-dark"> Order Summary</button>
+                    <div class="modal-bg-content">
+                        <div class="modal">
+                            <h1>Order Placed!</h1>
+                            <hr />
+                            <p class="p-lg">Congratulations! Your order has been confirmed.</p>
+                            <p class="p-lg">Please check your email for details.</p>
```
Comment (dev-ashishsethi) [inline docs/modals.html:84]: use class names that are easier to understand

File: stylesheets/components/modals/modal.css [hunk 7]
```diff
@@ -5,30 +5,45 @@
...
-    filter: blur(2px);
+    background-color: var(--PURE-WHITE);
 }
 
 .modal-cta-right {
     display: flex;
     justify-content: flex-end;
     padding: 4px 7px;
-    margin-top: 12px;
+    margin-top: 0.75rem;
```
Comment (dev-ashishsethi) [inline stylesheets/components/modals/modal.css]: for sizes less that 1 rem use pixels i didn't understand why did you remove 12px and put 0.75 rem instead

File: stylesheets/components/modals/modal.css [hunk 8]
```diff
@@ -5,30 +5,45 @@
...
 
+.modal-handler{
+    width: fit-content;
+    margin: auto;
+}
+
+.modal-bg-content{
+    display: none;
+    position: fixed;
+    z-index: 1;
```
Comment (dev-ashishsethi) [inline stylesheets/components/modals/modal.css:16]: you can give z-index values and sizes vales in variable it will make easier for you to modify your code in the future

**Review by dev-ashishsethi — APPROVED** [review]
do check comments

**Comment by sushmita12345** [comment]
Mostly suggestion are mentioned so go with the suggested changes.
You have followed mostly good practices, as use of % and utility class, everything looks good so you can proceed to merge.

**Review by sushmita12345 — APPROVED** [review]

File: docs/modals.html (hunk 2 above)
Comment (SinchanaMS) [reply docs/modals.html]: Thank you, I didn't think of this. Made the changes.

**Comment by SinchanaMS** [comment]
Thank you all for the reviews. I have made the necessary changes, merging to temp-dev.
